src/main/java/uk/nhs/tis/trainee/notifications/service/NotificationService.java

```diff
@@ -67,12 +89,37 @@ public void execute(JobExecutionContext jobExecutionContext) {
     Map<String, String> result = new HashMap<>();
     JobDataMap jobDetails = jobExecutionContext.getJobDetail().getJobDataMap();
 
-    log.info("Sent {} notification for {}", jobKey, jobDetails.getString(TIS_ID));
-    Instant processedOn = Instant.now();
-    result.put("status", "sent " + processedOn.toString());
-    jobExecutionContext.setResult(result);
-
-    saveNotificationSent(jobDetails, DUMMY_EMAIL, processedOn);
+    String personId = jobDetails.getString(PERSON_ID_FIELD);
+
+    UserAccountDetails userAccountDetails = getCognitoAccountDetails(personId);
+    if (userAccountDetails == null) {
+      //no Cognito details, so get from trainee-details API.
+      userAccountDetails = getTraineeDetailsAccountDetails(personId);
+    }
+
+    if (userAccountDetails != null) {
+      String programmeName = jobDetails.getString(PROGRAMME_NAME_FIELD);
+      LocalDate startDate = (LocalDate) jobDetails.get(START_DATE_FIELD);
+      NotificationType notificationType
+          = NotificationType.valueOf(jobDetails.get(NOTIFICATION_TYPE_FIELD).toString());
+      TisReferenceInfo tisReferenceInfo = new TisReferenceInfo(PROGRAMME_MEMBERSHIP,
+          jobDetails.get(TIS_ID_FIELD).toString());
+      jobDetails.putIfAbsent("name", userAccountDetails.familyName());
+      if (ACTUALLY_SEND_EMAIL) {
+        try {
+          emailService.sendMessage(personId, userAccountDetails.email(), notificationType,
+              templateVersion, jobDetails.getWrappedMap(), tisReferenceInfo);
+        } catch (MessagingException e) {
+          throw new RuntimeException(e);
```

Comment:
Here's the code smell. Because the overridden function doesn't throw a checked exception, the only obvious alternative would be annotating it with `@SneakyThrows` 🤔